src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -19,14 +21,16 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     internal sealed class FunctionPointerMethodSymbol : MethodSymbol
     {
         private readonly ImmutableArray<FunctionPointerParameterSymbol> _parameters;
+        private ImmutableHashSet<CustomModifier>? _lazyCallingConventionModifiers;
 
         public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSyntax syntax, Binder typeBinder, DiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved, bool suppressUseSiteDiagnostics)
         {
-            var callingConvention = getCallingConvention(syntax.CallingConvention, diagnostics);
+            ArrayBuilder<CustomModifier> customModifiers = ArrayBuilder<CustomModifier>.GetInstance();
+            var callingConvention = getCallingConvention(typeBinder, syntax.CallingConvention, customModifiers, diagnostics);
 
             RefKind refKind = RefKind.None;
             TypeWithAnnotations returnType;
-            var refReadonlyModifiers = ImmutableArray<CustomModifier>.Empty;
+            var refCustomModifiers = ImmutableArray<CustomModifier>.Empty;
```

Comment:
>var refCustomModifiers = ImmutableArray<CustomModifier>.Empty; [](start = 12, length = 62)

Consider moving this declaration closer to where this local can get a different value assigned.

Reply:
I can't, it needs to be on this level for use in the final `if`. This is as close as it can get.

---
In reply to: [457686746](https://github.com/dotnet/roslyn/pull/46110#discussion_r457686746) [](ancestors = 457686746)

Reply:
> I can't, it needs to be on this level for use in the final if. This is as close as it can get.

It looks like the first reference to this local, ```refCustomModifiers```, is on line 93. It feels like the declaration can be moved to line 90.


---
In reply to: [457780486](https://github.com/dotnet/roslyn/pull/46110#discussion_r457780486) [](ancestors = 457780486,457686746)